renpy/persistent.py

```diff
@@ -118,6 +113,12 @@ def _update(self):
                 "_seen_translates" : 0,
             }
 
+    def _keys(self):
+        """
+        Returns the set of keys which the creator should be aware of.
+        """
+        return [k for k in self.__dict__ if not k.startswith("_")]
```

Comment:
This function doesn't seem to add much besides materialisation of a list that doesn't ever need materialising.

Reply:
I suggest you read the opening message of the PR before making suggestions. The idea is to make it accessible for creators, either directly or through `__contains__`. And in the meantime, to factorize code.

Reply:
I did read it, but even if I did see the value in giving persistent a `__contains__` method (I don't, a key point of `persistent` is that unset attributes come back as `None`), adding a method that materialises a list isn't necessary to support it, or any thing else in this PR.

Reply:
A list was already created before my edits, so this is not a regression.
The reason I won't change it is because not creating a list (returning a parenthesized iterator for example) would yield a risk (in later separate uses of the function) that the contents change between the function call and the iteration, and/or related to the single-use nature of an iterator, which are not obvious bugs.

Reply:
I'm not suggesting that it should return an iterator, simply that the function doesn't need to exist at all.